lib/next_ls/check.ex

```diff
@@ -0,0 +1,30 @@
+defmodule NextLS.Check do
```

Comment:
i assume you got this terminology from the credo language server.

there its called a check because there are credo checks. i'm not sure if the same concept applies here.

I think the thing to do would have the elixir extension have the code for converting its diagnostics into code actions.

with that, i'm not sure exactly if a protocol is going to be the best way to go here, as they are all going to be  the same diagnostic struct.